uber/models.py

```diff
@@ -1938,6 +1953,22 @@ class ArbitraryCharge(MagModel):
     _repr_attr_names = ['what']
 
 
+class LedgerItem(MagModel):
+    ident = Column(UnicodeText, nullable=True)
+    type = Column(Choice(c.LEDGER_TYPE_OPTS), default=c.ACCOUNTS_RECEIVED)
+    method = Column(Choice(c.LEDGER_METHOD_OPTS), default=c.CC_STRIPE)
+    amount = Column(Integer)
+    when = Column(UTCDateTime, default=lambda: datetime.now(UTC))
+    who = Column(UnicodeText)
+    desc = Column(UnicodeText)
+
+    group_id = Column(UUID, ForeignKey('group.id', ondelete='SET NULL'), nullable=True)
+    group = relationship(Group, backref='ledger_items', foreign_keys=group_id, cascade='save-update,merge,refresh-expire,expunge')
+
+    attendee_id = Column(UUID, ForeignKey('attendee.id', ondelete='SET NULL'), nullable=True)
+    attendee = relationship(Attendee, backref='ledger_items', foreign_keys=attendee_id, cascade='save-update,merge,refresh-expire,expunge')
```

Comment:
I recommend a more flexible design like what we have in ``Tracking`` or ``Email``:

```python
fk_id = Column(UUID)
fk_model = Column(UnicodeText)
```

This way you can attach a ``LedgerItem`` onto literally any model even one which hasn't been thought of yet.

Assuming that you still want a property that makes it easy to get all the ledger items for a model instance, we can put something like this onto the ``MagModel`` class:

```python
@property
def ledger_items(self):
    return self.session.query(LedgerItem).filter_by(fk_id=self.id).all()
```

This approach will be more flexible in the future as we want to attach charges to more things.